docs/library/openamp.rst

```diff
@@ -0,0 +1,105 @@
+stopping a remote core, via the `RemoteProc` class.
+
+Example usage::
+
+    import openamp
+    
+    def ept_recv_callback(src, data):
+        print("Received message on endpoint", data)
+    
+    # Create a new RPMsg endpoint to communicate with the remote core.
+    ept = openamp.Endpoint("vuart-channel", callback=ept_recv_callback)
+    
+    # Create a remoteproc object, load its firmware and start it.
+    rproc = openamp.RemoteProc("virtual_uart.elf") # Or entry point address (ex 0x081E0000)
+    rproc.start()
+    
+    while True:
+        if ept.is_ready():
+            ept.send("data")
+
+Functions
+---------
+
+.. function:: new_service_callback(ns_callback)
+
+    Set the new service callback.
+
+        - ``ns_callback`` This function will be called back when the remote processor
+          announce new services. At the point the host processor can choose to create the
+          announced endpoint, if this particular service is supported, or ignore it if it's
+          not. If this function is not set, the host processor should first register the
+          endpoint locally, and it will be automatically bound when the remote announces
+          the service.
+
+Endpoint class
+--------------
+
+.. class:: Endpoint(name, callback, src=ENDPOINT_ADDR_ANY, dest=ENDPOINT_ADDR_ANY)
+
+   Construct a new RPMsg Endpoint. An endpoint is a bidirectional communication
+   channel between two cores.
+
+        - ``name`` The name of the endpoint.
+        - ``callback`` This function is called when the endpoint receives data with the source
+          address of the remote point, and the data as bytes passed by reference.
+        - ``src`` The endpoint source address. If none is provided, one will be assigned
+          to the endpoint by the library.
+        - ``dest`` The endpoint destination address. If the endpoint is created from the
+          new_service_callback, this must be provided and it must match the remote endpoint's
+          source address. If the endpoint is registered locally, before the announcement, the
+          destination address will be assigned by the library when the endpoint is bound.
+
+.. method:: Endpoint.deinit()
+
+   Destroy the endpoint and release all of its resources.
+
+.. method:: Endpoint.is_ready()
+
+   Returns True if the endpoint is ready to send (i.e., has both a source and destination addresses)
+
+.. method:: Endpoint.send(src=-1, dest=-1, timeout=0)
+
+   Send a message to the remote processor over this endpoint.
+
+        - ``src`` The source endpoint address of the message. If none is provided, the
+          source address the endpoint is bound to is used.
+        - ``dest`` The destination endpoint address of the message. If none is provided,
+          the destination address the endpoint is bound to is used.
+        - ``timeout`` Specifies the time in milliseconds to wait for a free buffer. By default
+          `send()` function is blocking.
```

Comment:
Does the default of `timeout=0` mean no blocking, or infinite blocking?  If the default is blocking forever then maybe the default value here should be `timeout=-1`.

Reply:
infinite blocking, and I've changed it to `-1` and signed int.